src/crystal/main.cr

```diff
@@ -47,6 +47,11 @@ module Crystal
     GC.init
 
     remember_blocking_state
+    at_exit do
```

Comment:
I am not totally convinced that handling the flush and restore of blocking state should be done with `at_exit`. I would rather leave some explicit calls for this. at_exit can't be cleared now, but might in the future. That way, also `@@handlers` would remain as `nil` in most of the cases.

Reply:
Maybe we should have multiple kind of at exit handlers? Like the app's at exit handlers with the user's code (which can be cleared), and the crystal's at exit handlers?

Reply:
nvm, it's just easier/faster to just duplicate the flushing+restore blocking, it's really simple code.

(was thinking of putting this in a class method in `Crystal` namespace (in `src/crystal/main.cr`) but I can't find a good name for it..)